cmd/expiry.go

```diff
@@ -0,0 +1,35 @@
+package cmd
+
+import (
+	"errors"
+	"regexp"
+	"strconv"
+	"time"
+)
+
+var nowFunc = time.Now
+
+func parseExpiry(fromNow string) (time.Time, error) {
+	re := regexp.MustCompile("\\s*(\\d+)\\s*(day|month|year|hour)s?")
+	matches := re.FindAllStringSubmatch(fromNow, -1)
+	addDate := map[string]int{
+		"day":   0,
+		"month": 0,
+		"year":  0,
+		"hour":  0,
+	}
+	for _, r := range matches {
+		addDate[r[2]], _ = strconv.Atoi(r[1])
```

Comment:
Why not check the error here?

Reply:
`r[1]` should have been matched from the first group in the regex which is `\d+` so we can say with fair certainty that it's going to be a contiguous string of integers.

Thinking about it though Atoi could still fail with an integer overflow (on e.g. "999999999999 days"). Should also probably check for signed int overflow because Atoi would not error in that case but the result is not valid in this context.

I will add those checks.